demisto_sdk/commands/common/clients/__init__.py

```diff
@@ -61,41 +64,54 @@ def get_client_from_marketplace(
         base_url: the base URL, if not provided will take from DEMISTO_BASE_URL env var
         api_key: the api key, if not provided will take from DEMISTO_API_KEY env var
         auth_id: the auth ID, if not provided will take from XSIAM_AUTH_ID env var
+        username: the username to authenticate, relevant only for xsoar on prem
+        password: the password to authenticate, relevant only for xsoar on prem
         verify_ssl: whether in each request SSL should be verified, True if yes, False if not
                     if verify_ssl = None, will take the SSL verification from DEMISTO_VERIFY_SSL env var
 
     Returns:
         the correct client according to the marketplace provided
     """
+    _base_url = base_url or os.getenv(DEMISTO_BASE_URL, "")
+    _api_key = api_key or os.getenv(DEMISTO_KEY, "")
+    _auth_id = auth_id or os.getenv(AUTH_ID)
+    _username = username or os.getenv(DEMISTO_USERNAME, "")
+    _password = password or os.getenv(DEMISTO_PASSWORD, "")
+    _verify_ssl = (
+        verify_ssl
+        if verify_ssl is not None
+        else string_to_bool(os.getenv(DEMISTO_VERIFY_SSL, False))
+    )
+
     if marketplace in (MarketplaceVersions.XSOAR_ON_PREM, MarketplaceVersions.XSOAR):
         config = XsoarClientConfig(
-            base_api_url=base_url, api_key=api_key, verify_ssl=verify_ssl
+            base_api_url=_base_url, api_key=_api_key, verify_ssl=_verify_ssl
         )
     elif marketplace == MarketplaceVersions.XSOAR_SAAS:
         config = XsoarSaasClientConfig(
-            base_api_url=base_url,
-            api_key=api_key,
-            auth_id=auth_id,
-            verify_ssl=verify_ssl,
+            base_api_url=_base_url,
+            api_key=_api_key,
+            auth_id=_auth_id,
+            verify_ssl=_verify_ssl,
         )
     else:
         config = XsiamClientConfig(
-            base_api_url=base_url,
-            api_key=api_key,
-            auth_id=auth_id,
-            verify_ssl=verify_ssl,
+            base_api_url=_base_url,
+            api_key=_api_key,
+            auth_id=_auth_id,
+            verify_ssl=_verify_ssl,
         )
     return get_client_from_config(config)
 
 
 @lru_cache
 def get_client_from_server_type(
-    base_url: Optional[str] = os.getenv(DEMISTO_BASE_URL, ""),
```

Comment:
why move this into the function body?

Reply:
because when we define default values, python automatically reads them if the module that is being imported in some sort of way, that can cause un-expected issues, for example we might read env vars before reading `.env` file, that is just more secure way of doing it.